packages/braintree-utils/src/braintree-script-loader.ts

```diff
@@ -46,103 +47,126 @@ export default class BraintreeScriptLoader {
-        return this.loadBraintreeModuleOrThrow<BraintreeUsBankAccountCreator>(
-            BraintreeModuleName.usBankAccount,
-            'us-bank-account.min.js',
+        return this.loadScript<BraintreeUsBankAccountCreator>(
+            BraintreeModuleName.UsBankAccount,
+            BraintreeSdkFileName.UsBankAccount,
         );
     }
 
     async loadGooglePayment(): Promise<GooglePayCreator> {
-        return this.loadBraintreeModuleOrThrow<GooglePayCreator>(
-            BraintreeModuleName.googlePayment,
-            'google-payment.min.js',
+        return this.loadScript<GooglePayCreator>(
+            BraintreeModuleName.GooglePayment,
+            BraintreeSdkFileName.GooglePayment,
         );
     }
 
     async load3DS(): Promise<BraintreeThreeDSecureCreator> {
-        return this.loadBraintreeModuleOrThrow<BraintreeThreeDSecureCreator>(
-            BraintreeModuleName.threeDSecure,
-            'three-d-secure.min.js',
+        return this.loadScript<BraintreeThreeDSecureCreator>(
+            BraintreeModuleName.ThreeDSecure,
+            BraintreeSdkFileName.ThreeDSecure,
         );
     }
 
     loadVisaCheckout(): Promise<BraintreeVisaCheckoutCreator> {
-        return this.loadBraintreeModuleOrThrow<BraintreeVisaCheckoutCreator>(
-            BraintreeModuleName.visaCheckout,
-            'visa-checkout.min.js',
+        return this.loadScript<BraintreeVisaCheckoutCreator>(
+            BraintreeModuleName.VisaCheckout,
+            BraintreeSdkFileName.VisaCheckout,
         );
     }
 
     loadVenmoCheckout(): Promise<BraintreeVenmoCheckoutCreator> {
-        return this.loadBraintreeModuleOrThrow<BraintreeVenmoCheckoutCreator>(
-            BraintreeModuleName.venmo,
-            'venmo.min.js',
+        return this.loadScript<BraintreeVenmoCheckoutCreator>(
+            BraintreeModuleName.Venmo,
+            BraintreeSdkFileName.Venmo,
         );
     }
 
     async loadHostedFields(): Promise<BraintreeHostedFieldsCreator> {
-        return this.loadBraintreeModuleOrThrow<BraintreeHostedFieldsCreator>(
-            BraintreeModuleName.hostedFields,
-            'hosted-fields.min.js',
+        return this.loadScript<BraintreeHostedFieldsCreator>(
+            BraintreeModuleName.HostedFields,
+            BraintreeSdkFileName.HostedFields,
+        );
+    }
+
+    private async loadScript<T extends BraintreeModuleCreators>(
+        moduleName: BraintreeModuleName,
+        fileName: BraintreeSdkFileName,
+    ): Promise<T> {
+        const hash = this.getIntegrityValuesByFileName(fileName)[this.braintreeSdkVersion];
+
+        return this.loadBraintreeModuleOrThrow<T>(
+            moduleName,
+            fileName,
+            hash
+                ? {
+                      async: true,
+                      attributes: {
+                          integrity: hash,
+                          crossorigin: 'anonymous',
+                      },
+                  }
+                : undefined,
```

Comment:
It is not necessary to create another method for this... you can update loadBraintreeModuleOrThrow instead

Reply:
Or you can probably create a method called `getScriptOptions` and pass a `moduleName` inside and also use this method in `loadBraintreeModuleOrThrow`